pcdsdevices/gon.py

```diff
@@ -550,6 +536,36 @@ def inverse(self, real_pos):
                                           real_pos.phi)
         return self.PseudoPosition(e_eta=e_eta, e_chi=e_chi, e_phi=e_phi)
 
+    @pseudo_position_argument
+    def move(self, position, wait=True, timeout=None, moved_cb=None):
+        """
+        Move to a specified position, optionally waiting for motion to
+        complete.
+
+        Checks for the motor step, and ask the user for confirmation if
+        movement step is greater than default one.
+        """
+        try:
+            return super().move(position, wait=wait, timeout=timeout,
+                                moved_cb=moved_cb)
+        except KappaMoveAbort as exc:
+            logger.warning('Aborting moving for safety.')
+            status = DeviceStatus(self)
+            status.set_exception(exc)
+            return status
+
+    def check_value(self, position):
+        """
+        Check the goal position and raise an exception if it is not safe.
+
+        This is called before executing any move.
+        """
+        super().check_value(position)
+        eta, kappa, phi = self.e_to_k(position.e_eta, position.e_chi,
```

Comment:
@ZLLentz  and @dlzhu  - Thanks for fixing this - I think this was my mistake for not using the conversion here

Also, I did not think of `check_value` for this solution, that's great - thanks

Reply:
Moving to `check_value` wasn't entirely necessary, but since we were here I thought it was the canonical place to locate the check.